src/lib.rs

```diff
@@ -119,6 +121,42 @@ fn bitfield_inner(args: TokenStream, input: TokenStream) -> syn::Result<TokenStr
         quote!()
     };
 
+    let defmt_impl = if defmt {
```

Comment:
This function is getting quite long (especially if we want to support different formatters).
I would recommend moving the formatting code (including `core::fmt`) into different functions (including the parts inside `Member::debug` / `Member::defmt`).
Something alongside the following would be nice at the top level:

```rust
let debug_impl = implement_debug(debug, name, &members);
```

`debug` is an enum (like `Debug::Defmt`/`Debug::Core`) as explained earlier.

Reply:
Do you mean inlining the existing `Member::debug`/`Member::defmt` inside this new `implement_debug` function?

Reply:
Yes, then the format implementation are mostly self contained.